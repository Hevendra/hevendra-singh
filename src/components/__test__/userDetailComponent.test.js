import React from "react";
import { shallow } from "enzyme";
import UserDetail from "../userDetailComponent";

describe("<UserDetailComponent/>", () => {
  const wrapper = shallow(
    <UserDetail
      key={1}
      data={{
        first_name: "George",
        last_name: "Edwards",
        avatar:
          "https://s3.amazonaws.com/uifaces/faces/twitter/mrmoiree/128.jpg"
      }}
    />
  );

  it("should render user information correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render correct user name", () => {
    expect(wrapper.find(".display-name").text()).toEqual("George Edwards ");
  });

  it("should render correct user image", () => {
    expect(wrapper.find("img").props().src).toEqual(
      "https://s3.amazonaws.com/uifaces/faces/twitter/mrmoiree/128.jpg"
    );
  });
});
