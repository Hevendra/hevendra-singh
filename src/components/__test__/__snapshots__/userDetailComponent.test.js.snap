// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDetailComponent/> should render user information correctly 1`] = `
<div
  className="card-body"
>
  <div
    className="wrapper"
  >
    <img
      alt="PS"
      className="image--cover"
      id="image_avatar"
      src="https://s3.amazonaws.com/uifaces/faces/twitter/mrmoiree/128.jpg"
    />
  </div>
  <p
    className="display-name"
  >
    George Edwards
     
  </p>
  <p
    className="button"
    onClick={[Function]}
  >
    <span>
      Delete 
    </span>
  </p>
</div>
`;
